src/fstab-generator/fstab-generator.c

```diff
@@ -1510,20 +1531,25 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         } else if (streq(key, "usrhash")) {
 
+                if (generator_soft_rebooted())
+                        return 0;
+
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
                 return free_and_strdup_warn(&arg_usr_hash, empty_to_null(value));
 
-        } else if (streq(key, "rw") && !value)
+        } else if (streq(key, "rw") && !value && !generator_soft_rebooted())
                 arg_root_rw = true;
-        else if (streq(key, "ro") && !value)
+        else if (streq(key, "ro") && !value && !generator_soft_rebooted())
                 arg_root_rw = false;
         else if (streq(key, "systemd.volatile")) {
-                VolatileMode m;
+
+                if (generator_soft_rebooted())
```

Comment:
This means, even if booted with `systemd.volatile`, disk will be writable after soft-reboot. I think it is dangerous. No?